modules/navier_stokes/doc/content/source/kernels/INSADConvectedMesh.md

```diff
@@ -0,0 +1,21 @@
+# INSADConvectedMesh
+
+!syntax description /Kernels/INSADConvectedMesh
+
+`INSADConvectedMesh` implements the corresponding weak form for the components of
+the term:
+
+\begin{equation}
+-\rho \left(\frac{\partial\vec{d_m}}{\partial t} \cdot \nabla\right) \vec{u}
+\end{equation}
+
+where $\rho$ is the density, $\vec{d_m}$ is the fluid mesh displacements, and
+$\vec{u}$ is the fluid velocity. This term is essential for obtaining the
+correct convective derivative of the fluid in cases where the fluid mesh is
+dynamic, e.g. in simulations of fluid-structure interaction.
```

Comment:
I don't think so because you can do these fuzzy fluid/solid domain approximations. And even if you take the stance that we shouldn't do these fuzzy treatments, I think there's a valid argument for having the object in this module since it adds to the NS momentum equation

Reply:
ah I meant to delete that comment but I did not catch it.
This is great in NS